next-auth-with-directus/index.md

```diff
@@ -0,0 +1,1040 @@
+
+## Building the Frontend Application (Nextjs)
+
+With the backend already running, you can now start building the application's frontend. To do so, first navigate to the _next-directus_ folder if you are still in the _server_ folder with the command:
+
+```bash
+cd ../
+```
+
+### Installing Nextjs
+
+Inside of the _next-directus_ folder, run the following command to install nextjs
+
+```bash
+npx create-next-app@latest client
+```
+
+During installation, when prompted, choose the following configurations:
+
+```bash
+✔ Would you like to use TypeScript?  Yes
+✔ Would you like to use ESLint?  Yes
+✔ Would you like to use Tailwind CSS? Yes
+✔ Would you like to use `src/` directory?  No
+✔ Would you like to use App Router? (recommended)  Yes
+✔ Would you like to customize the default import alias (@/*)? Yes
+✔ What import alias would you like configured? @/*
+
+```
+
+This will install Nextjs with Typescript and TailwindCSS configurations that are ready to use.
+
+Navigate into the _client_ folder with the command:
+
+```bash
+cd client
+```
+
+Start the Nextjs server with the command below. This will open up the default Nextjs page template for a new project.
+
+```bash
+npm run dev
+```
+
+You should have a Nextjs application running in <http://localhost:3000/>
+
+### Installing the required dependencies
+
+For the frontend of the application, you need the following dependencies:
+
+- [Next Auth](https://next-auth.js.org/) for creating the authentication system for Nextjs
+- [Directus JavaScript SDK](https://docs.directus.io/guides/sdk/getting-started.html) for communicating with your Directus backend.
+
+Install the dependencies using the command:
+
+```bash
+npm i next-auth @directus/sdk
+```
+
+### Configuring the Dependencies
+
+Create a _.env.local_ with the following contents:
+
+`.env.local`
+
+```bash
+NEXTAUTH_URL=http://localhost:3000
+NEXTAUTH_SECRET=YOUR_NEXT_RANDOM_SECRET
+USER_ROLE=THE_ROLE_OF_YOUR_CUSTOMER_FROM_DIRECTUS
+BACKEND_URL=http://localhost:8055 #YOUR DIRECTUS BACKEND URL
+NEXT_PUBLIC_URL=http://localhost:3000
+
+```
+
+This _.env.local_ contains all the credentials needed by NextAuth and the Directus SDK.
+
+### Configuring the Directus SDK
+
+Create a new folder called _lib_, and inside of that folder, create a new file called _directus.ts_ with the following contents:
+
+`lib/directus.ts`
+
+```ts
+import { createDirectus, rest, authentication } from '@directus/sdk';
+
+const directus = createDirectus(BACKEND_URL).with(authentication("cookie", {credentials: "include", autoRefresh: true})).with(rest());
+
+export default directus;
+```
+
+Here, you are creating a new Directus API instance with enabled `authentication` and `rest` API features.
+
+### Creating the AuthForm Component
+
+With all the required dependencies installed, you can start building the components for the frontend application.
```

Comment:
Remove this. It's clear we've moved on from boilerplating to implementation - you don't need to be as explicit as to call it out.